airflow/www/templates/admin/master.html

```diff
@@ -128,64 +128,75 @@
    return readableFormat
 }
 
-// Capture keyboard shortcuts
-$(document).keyup(function(e) {
-  if (!/^(?:input|textarea|select|button)$/i.test(e.target.tagName)) {
-    let key = e.which || e.keyCode;
-    let letterT = 84;
-    if (key === letterT) {
-      initTaskSearchModal();
-    }
+// Get task search data in the background
+$.getJSON("{{ url_for('airflow.all_tasks') }}", function (data) {
+  var typeahead_data = [];
+  for (const [dag, tasks] of Object.entries(data)) {
+    typeahead_data = typeahead_data.concat(tasks.flatMap(task => dag + " " + task));
```

Comment:
actually this should be a list of objects like 
```json
[
  {"dag_id":"blah1", "task_id":"blah1", "name":"blah1 blah1"},
  {"dag_id":"blah2", "task_id":"blah2", "name":"blah2 blah2"},
]
```

Reply:
looking up online, concat should work ?

Reply:
yeah it's more that the flatmap returns a string instead of an object

Reply:
ah right. you need map()

Reply:
i think just `typeahead_data = typeahead_data.concat(tasks.flatMap(task => {"dag_id": dag, "task_id": task, "name": dag + " " + task));`?